File: api/app/Models/KycTimeline.php
<?php

namespace App\Models;

use App\Models\Scopes\ApplicantIndividualCompanyIdScope;
use App\Models\Traits\BaseObServerTrait;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\MorphTo;

class KycTimeline extends BaseModel
{
    use BaseObServerTrait;

    public $table = 'kyc_timeline';

    public $timestamps = false;

    protected $fillable = [
        'creator_id',
        'os',
        'browser',
        'ip',
        'action',
        'tag',
        'action_type',
        'document_id',
        'company_id',
        'applicant_id',
        'applicant_type',
        'action_old_value',
        'action_new_value',
        'created_at',
    ];

    protected $casts = [
        'action_old_value' => 'array',
        'action_new_value' => 'array',
        'created_at' => 'datetime:YYYY-MM-DDTHH:mm:ss.SSSZ',
    ];

    protected static function booted()
    {
        parent::booted();
        static::addGlobalScope(new ApplicantIndividualCompanyIdScope());
    }

    public function company(): BelongsTo
    {
        return $this->belongsTo(Company::class);
    }

    public function clientable(): MorphTo
    {
        return $this->morphTo(__FUNCTION__, 'applicant_type', 'applicant_id');
    }

    public function document(): BelongsTo
    {
        return $this->belongsTo(ApplicantDocument::class);
    }

    public function member(): BelongsTo
    {
        return $this->belongsTo(Members::class, 'creator_id', 'id');
    }
}

File: go-auth/models/postgres/kyc_timeline.go
package postgres

import "time"

type KycTimeLine struct {
	ID            uint64    `gorm:"primarykey,column:id"`
	CreatorId     uint64    `gorm:"column:creator_id"`
	Os            string    `gorm:"column:os"`
	Browser       string    `gorm:"column:browser"`
	Ip            string    `gorm:"column:ip"`
	Tag           string    `gorm:"column:tag"`
	Action        string    `gorm:"column:action"`
	ActionType    string    `gorm:"column:action_type"`
	DocumentId    uint64    `gorm:"column:document_id"`
	CompanyId     uint64    `gorm:"column:company_id"`
	ApplicantId   uint64    `gorm:"column:applicant_id"`
	ApplicantType string    `gorm:"column:applicant_type"`
	CreatedAt     time.Time `gorm:"column:created_at"`
}

func (*KycTimeLine) TableName() string {
	return "kyc_timeline"
}
